public/app/plugins/panel/heatmap-new/HeatmapPanel.tsx

```diff
@@ -97,6 +108,8 @@ export const HeatmapPanel: React.FC<HeatmapPanelProps> = ({
   dataRef.current = info;
 
   const builder = useMemo(() => {
+    const scaleConfig = dataRef.current?.heatmap?.fields[1].config?.custom
```

Comment:
@leeoniya I changed this so we read the scale from DataFrame so the same thing can be used regardless where it came from